staging/src/k8s.io/client-go/tools/cache/fifo.go

```diff
@@ -328,12 +381,11 @@ func (f *FIFO) Replace(list []interface{}, resourceVersion string) error {
 		f.initialPopulationCount = len(items)
 	}
 
+	// replace queue and items
+	f.queue = queue
```

Comment:
prior f.queue needs to be cleared? would it leak memory otherwise?

Reply:
yes, its a good question. using 
```f.queue.Init()```
before replacing maybe a good way where func `Init` initializes or clears list. see https://pkg.go.dev/container/list#List.Init